ftw/publisher/core/tests/test_portlet_adapter.py

```diff
@@ -208,3 +208,27 @@ def test_portlets_with_settings(self):
             IPortletAssignmentSettings.providedBy(settings),
             'Portlet settings is not PortletAssignmentSettings but %s' % (
                 type(settings)))
+
+    def test_portlet_visibility_settings(self):
+        assignment = self.layer['right_portlets'].get('title2', False)
+        settings = {'visible': True}
+        IPortletAssignmentSettings(assignment).data = settings
+
+        # get the data
+        adapter = getAdapter(self.layer['folder1'], IDataCollector,
+                             name="portlet_data_adapter")
+        data = adapter.getData()
```

Comment:
Could you run that through `json`? e.g.:
```python
data = adapter.getData()
data = json.loads(json.dumps(data))
```

The reason is that we require to have json serializable data so that we can transport it.
By running it through `json` we would detect things such as persistentdict.